streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidEventMapper.java

```diff
@@ -55,20 +57,27 @@ public DruidEvent(StreamlineEvent event) {
             this.event = event;
         }
 
-        public static boolean isMap(Object map) {
-            return map instanceof Map;
-        }
-
         @Override
         protected Map<String, Object> delegate() {
-            return event;
+            Map<String, Object> updatedEvent = new LinkedHashMap<>();
+            event.forEach((key, value) -> {
+                flattenNestedFields(key, value, updatedEvent, new LinkedList<>(Arrays.asList(key)));
+            });
+            LOG.debug("updated DruidEvent: {}", updatedEvent );
+            return updatedEvent;
         }
 
-        @Override
-        public Object get(@Nullable Object key) {
-            Object value = StreamlineEventUtils.getFieldValue(event, (String)key);
-            LOG.debug("called for key: {}, value: {}", key, value);
-            return value;
+        private void flattenNestedFields(String key, Object value, Map<String, Object> updatedEvent, List<String> parentKeyList) {
+            if (value instanceof  Map) {
+                Map<String, Object> subMap = (Map<String, Object>) value;
+                subMap.forEach((childKey, childValue) -> {
+                    List<String> updatedParentKeyList = new LinkedList<>(parentKeyList);
```

Comment:
you could make the `parentKey` a string and then directly append the childKey and pass it around to avoid creating the new list.

e.g. 

```java
subMap.forEach((childKey, childValue) -> {
  flattenNestedFields(childKey, childValue, updatedEvent, parentKey + "." + childKey);
});
```